src/components/Data/Documents/Page.vue

```diff
@@ -411,7 +413,10 @@ export default {
           this.index,
           this.collection
         )
-        this.fetchDocuments()
+        await waitForConnected(1000)
+        if (Vue.prototype.$kuzzle.protocol.state === 'connected') {
```

Comment:
You don't need to get `$kuzzle` from `Vue.prototype` as all the components are instances of the `Vue` class. You just need to do
```suggestion
        if (this.$kuzzle.protocol.state === 'connected') {
```